src/main/resources/grammar.hgr

```diff
@@ -41,13 +39,13 @@ grammar {
 
     # TODO: these should probably be partials...
```

Comment:
If a partial is a thing that would prevent duplication of the now-hairy regexps, then yes.  :smile: